src/components/KTextArea/KTextArea.vue

```diff
@@ -70,123 +70,73 @@
 </template>
 
 <script lang="ts">
-import { ref, computed, watch, useAttrs, useSlots, useId } from 'vue'
-import useUtilities from '@/composables/useUtilities'
-import KLabel from '@/components/KLabel/KLabel.vue'
-import type { TextAreaLimitExceed } from '@/types'
 import { cssSupports, getScrollbarSize, IS_MAYBE_FIREFOX } from '@/utilities/browser'
 
 const DEFAULT_CHARACTER_LIMIT = 2048
 // Firefox has the double click feature built in, so we don't need to calculate the size
 const RESIZE_HANDLE_SIZE = IS_MAYBE_FIREFOX ? 0 : getScrollbarSize()
 const SUPPORT_FIELD_SIZING_CONTENT = cssSupports('field-sizing', 'content')
-
-export default {
-  inheritAttrs: false,
-}
 </script>
 
 <script lang="ts" setup>
-const props = defineProps({
-  modelValue: {
-    type: String,
-    default: '',
-  },
-  label: {
-    type: String,
-    default: '',
-  },
-  overlayLabel: {
-    type: Boolean,
-    default: false,
-  },
-  labelAttributes: {
-    type: Object,
-    default: () => ({}),
-  },
-  characterLimit: {
-    type: [Boolean, Number],
-    default: DEFAULT_CHARACTER_LIMIT,
-    // Ensure the characterLimit is greater than zero
-    validator: (limit: number | boolean): boolean => typeof limit === 'number' ? limit > 0 : true,
-  },
-  rows: {
-    type: Number,
-    default: 5,
-  },
-  error: {
-    type: Boolean,
-    default: false,
-  },
-  resizable: {
-    type: Boolean,
-    default: false,
-  },
-  autosize: {
-    type: Boolean,
-    default: false,
-  },
-  help: {
-    type: String,
-    default: '',
-  },
-  /**
-   * @deprecated in favor of `resizable` prop
-   */
-  isResizable: {
-    type: Boolean,
-    default: false,
-    validator: (value: boolean): boolean => {
-      if (value) {
-        console.warn('KTextArea: the `isResizable` prop is deprecated in favor of the `resizable` prop. See the migration guide for more details: https://kongponents.konghq.com/guide/migrating-to-version-9.html#ktextarea')
-      }
-
-      return true
-    },
-  },
-  /**
-   * @deprecated in favor of `error` prop
-   */
-  hasError: {
-    type: Boolean,
-    default: false,
-    validator: (value: boolean): boolean => {
-      if (value) {
-        console.warn('KTextArea: the `hasError` prop is deprecated in favor of the `error` prop. See the migration guide for more details: https://kongponents.konghq.com/guide/migrating-to-version-9.html#ktextarea')
-      }
+import { ref, computed, watch, useAttrs, useId } from 'vue'
```

Comment:
Do we still actually need the two script sections now?

Reply:
I put some feature detection code in a separate `<script>` block so that they are evaluated at top level instead of repeatedly evaluated in every `setup` call.